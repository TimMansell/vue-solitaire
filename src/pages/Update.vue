<template>
  <GameOverlay center-content show-logo data-test="update">
    <template #title>
      <span v-if="isOldVersion" data-test="update-title">New Update</span>
      <span v-if="!isOldVersion" data-test="update-title">No New Updates</span>
    </template>
    <template #msg>
      <p v-if="isOldVersion">A new version of the game is available</p>
      <p v-if="!isOldVersion">You are on the latest version</p>
    </template>
    <template #buttons>
      <UpdateButton v-if="isOldVersion" />
      <ContinueGameButton v-if="!isOldVersion" />
    </template>
  </GameOverlay>
</template>

<script>
import { mapGetters } from 'vuex';
import GameOverlay from '@/components/GameOverlay.vue';
import UpdateButton from '@/components/UpdateButton.vue';
import ContinueGameButton from '@/components/ContinueGameButton.vue';

export default {
  name: 'UpdatePage',
  components: {
    GameOverlay,
    UpdateButton,
    ContinueGameButton,
  },
  computed: {
    ...mapGetters(['isOldVersion']),
  },
  beforeRouteLeave(to, from, next) {
    const { isOldVersion } = this;

    if (isOldVersion) {
      next(false);
    } else {
      next();
    }
  },
};
</script>
