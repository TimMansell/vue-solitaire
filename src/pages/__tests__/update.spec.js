import { shallowMount } from '@vue/test-utils';
import Update from '@/pages/Update.vue';

describe('Update.vue', () => {
  it('matches updates snapshot', () => {
    const wrapper = shallowMount(Update);

    expect(wrapper).toMatchSnapshot();
  });
});
