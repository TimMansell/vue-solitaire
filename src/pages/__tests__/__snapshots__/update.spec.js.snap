// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update.vue matches updates snapshot 1`] = `
<gameoverlay-stub centercontent="true" showlogo="true" data-test="update"><span>New Update</span>
  <p>A new version of the game is available</p>
  <updatebutton-stub></updatebutton-stub>
</gameoverlay-stub>
`;
